railties/test/generators/app_generator_test.rb

```diff
@@ -941,6 +941,14 @@ def test_system_tests_directory_generated
     assert_directory("test/system")
   end
 
+  unless Gem.win_platform?
+    def test_master_key_is_only_readable_by_the_owner
+      run_generator
+
+      assert_nil File.world_readable?("config/master.key")
```

Comment:
Since we're setting the group read bit as well we should make this more explicit like:

``` ruby
stat = File.stat("config/master.key")
assert_equal "100600", sprintf("%o, stat.mode)
```